### DISCUSSION TITLE
Graph reports

### DISCUSSION THREAD

**Discussion by LucaCappelletti94** [snap-stanford/ogb#434] [General]
Hi!

I have started running some experiments with OGB graphs, and I have noticed some of them include some topological oddities (isomorphic nodes, dendritic trees etc). I am not sure whether these are problematic for the task, as it is generally extremely task-dependent.

That being said, [using GRAPE utilities for graph report](https://github.com/AnacletoLAB/grape/blob/main/tutorials/Create%20extensive%20knowledge%20graph%20reports%20using%20GRAPE.ipynb) I can get you for all graphs rather quickly extensive reports, such as the one shown in the linked example.

These reports are generated in HTML (or Markdown) and can be readily dropped into the OGB website if desired - I believe they may offer insights to OGB users regarding the graph topology, and why their model is performing well / not so well on a given graph.

[Here is another example of a graph report, computed on the PubMed citation graph](https://docs.google.com/document/d/1hL6E4-NFODSQiwN8C0HC626_axnYTgPGgh-n4og_CNc/edit?usp=sharing).

If you believe these may be useful I can start providing them as part of a pull request, just let me know if you believe that to be the case. 

Best,
Luca

**Comment by weihua916** [comment]
Hi Luca, thanks for the pointer. I think this is definitely an interesting resource, but I do not feel necessary at the moment. I think users can feel free to use this resource if they want.

**Reply by LucaCappelletti94** [reply]
Ok!
